src/main/java/net/uaznia/lukanus/hudson/plugins/gitparameter/GitParameterDefinition.java

```diff
@@ -236,6 +240,15 @@ public void setBranchFilter(String branchFilter) {
         this.branchFilter = branchFilter;
     }
 
+    public String getListSize() {
+        return listSize;
```

Comment:
You must check listSize == null and return DEFAULT_LIST_SIZE because after install your patch (and I didn't do reconfiguration job) i got 
![image](https://user-images.githubusercontent.com/4460010/36646896-307cbade-1a7e-11e8-905b-fc63f6bee8d1.png)

Reply:
listSize does get initialized in the constructor.  I am guessing you had an existing job and then upgraded the plugin containing my changes?  This is a scenario I thought I tested...
Does this create a unique condition when a parameter is added to a plugin?   I am trying to understand how listSize would not have a value of 5.

Reply:
Exactly, If you have existing job constructor is not invoked. In this case listSize is null (not initialised)
Constructor is invoked only when configuration is saving

Reply:
Thanks for the feedback.  So for an existing job the DataBoundConstructor is not invoked? only the default constructor?  If that is the case would it make sense to define a default constructor or would that impede the function of the DataBoundContructor?

I primarily work in C/C++/Assembly and it has been quite a while (15 years at least) since I have done anything in Java.

Reply:
@DataBoundConstructor and @DataBoundSetter are invoke when you save configuration. First is search and invoke constructor which has DataBoundConstructor adnotation after this was search and invoke method which has @DataBoundSetter anottation.

Use @DataBoundSetter  It is excessive in this case But he can stay.